src/effects/RealtimeEffectManager.cpp

```diff
@@ -65,25 +66,31 @@ void RealtimeEffectManager::Initialize(double rate)
    // (Re)Set processor parameters
    mChans.clear();
    mRates.clear();
+   mGroupLeaders.clear();
 
    // RealtimeAdd/RemoveEffect() needs to know when we're active so it can
    // initialize newly added effects
    mActive = true;
 
-   // Tell each effect to get ready for action
+   // Tell each effect of the master list to get ready for action
    VisitGroup(nullptr, [rate](RealtimeEffectState &state, bool){
       state.Initialize(rate);
    });
 }
 
-void RealtimeEffectManager::AddTrack(int group, unsigned chans, float rate)
+void RealtimeEffectManager::AddTrack(Track *track, unsigned chans, float rate)
 {
-   VisitGroup(nullptr, [&](RealtimeEffectState &state, bool){
-      state.AddTrack(group, chans, rate);
-   });
-
-   mChans.push_back(chans);
-   mRates.push_back(rate);
+   auto leader = *track->GetOwner()->FindLeader(track);
```

Comment:
Here is the problem which I outlined during our conversation:

If I have a project with two tracks, the Visitor function will make the RealtimeEffectState be initialized a second time, thereby forgetting the indexes in the EffectProcessor slaves; this means that when play is pressed, both tracks will be processed by the same slave.

Reply:
My recent force push should fix this:  please confirm.

Reply:
I built PR 2500 and tested it with my usual test (two mono tracks and BassTreble used in realtime).
There seems to be a defect there which prevents me to test the fix (I will place a comment there to describe it)
Anyway, I have noticed that the code changed so much, that my initial comment is not applicable anymore, so I will press resolve and being this the last issue, I will approve this PR.